sdk/core/azure-core/test/ut/operation.cpp

```diff
@@ -0,0 +1,56 @@
+// ************ CUSTOMER CODE **********************
+
+#include "gtest/gtest.h"
+
+#include "operation_test.hpp"
+
+#include <azure/core/context.hpp>
+#include <azure/core/operation.hpp>
+
+#include <chrono>
+
+using namespace Azure::Core::Test;
+using namespace std::literals;
+
+TEST(Operation, Poll)
+{
+  StringClient client;
+  auto operation = client.StartStringUpdate();
+
+  EXPECT_FALSE(operation.Done());
+  EXPECT_FALSE(operation.HasValue());
+
+  while(!operation.Done())
+  {
+    EXPECT_FALSE(operation.HasValue());
+    auto response = operation.Poll();
+  }
+
+  EXPECT_TRUE(operation.Done());
+  EXPECT_TRUE(operation.HasValue());
+
+  auto result = operation.Value();
+  EXPECT_TRUE(result == "StringOperation-Completed");
+}
+
+TEST(Operation, PollUntilDone)
```

Comment:
Need `OperationStatus` tests, including equality comparison.

Reply:
Also need `GetResumeToken` tests.

Reply:
The equality comparison is implicitly tested inside of Done().
m_status == OperationStatus::Succeeded

I'll add some specific testing,